### PR TITLE
docs: Added troubleshooting guide

### LINKED ISSUE #5045: Create Dagger troubleshooting guide
### What is the issue?

Create a guide to troubleshoot Dagger.

Potentially this could also be a guide to troubleshoot Dagger pipelines.

Inspired by https://discord.com/channels/707636530424053791/1101085612750143568

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit adds a troubleshooting guide to the Dagger docs.

Closes #5045

### CODE REVIEW

File: docs/current/guides/235290-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+A Dagger pipeline may hang or become unresponsive, eventually generating a BuildKit error such as `buildkit failed to respond` or `container state improper`.
+
+To resolve this error, you must stop and remove the Dagger Engine container and clear unused volumes and the cache:
+
+1. Stop and remove the Dagger Engine container:
+
+  ```shell
+  DAGGER_CONTAINER_NETWORK_NAME=`docker ps --filter "name=^dagger-engine-*" --format '{{.Names}}'`
+  docker stop $DAGGER_CONTAINER_NETWORK_NAME
```
Comment (moljac024) [inline docs/current/guides/235290-troubleshooting.md:27]: Could we make this docker-agnostic or is it implied that the audience will know to substitute `docker` for `podman` if required?

Comment (vikram-dagger) [reply docs/current/guides/235290-troubleshooting.md:27]: Good question. WDYT @gerhard @kpenfound ?

Comment (gerhard) [reply docs/current/guides/235290-troubleshooting.md:27]: By default, Dagger relies on Docker for provisioning the Engine, so using `docker` in this context seems right to me. Resolving this. Please re-open if you disagree @moljac024.

I see that we have added a note about custom container runtimes. To me, this is more confusing than helpful. Will address that separately.

File: docs/current/guides/235290-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+A Dagger pipeline may hang or become unresponsive, eventually generating a BuildKit error such as `buildkit failed to respond` or `container state improper`.
+
+To resolve this error, you must stop and remove the Dagger Engine container and clear unused volumes and the cache.
+
+1. Stop and remove the Dagger Engine container:
+
+  ```shell
+  DAGGER_CONTAINER_NETWORK_NAME=`docker ps --filter "name=^dagger-engine-*" --format '{{.Names}}'`
+  docker stop $DAGGER_CONTAINER_NETWORK_NAME
+  docker rm $DAGGER_CONTAINER_NETWORK_NAME
```
Comment (gerhard) [inline docs/current/guides/235290-troubleshooting.md:28]: Technically, this is the Docker Engine container provisioned via Docker. I would rename this variable to `DAGGER_ENGINE_DOCKER_CONTAINER`. 

I would also use `"$()"` instead of `` for the command. Rationale: https://github.com/koalaman/shellcheck/wiki/SC2006#rationale

Instead of using `docker` subcommand aliases, I would use the namespace equivalent, e.g. `docker container ...`

All put together:

```console
DAGGER_ENGINE_DOCKER_CONTAINER="$(docker container list --all --filter 'name=^dagger-engine-*' --format '{{.Names}}')"
docker container stop "$DAGGER_ENGINE_DOCKER_CONTAINER"
docker container rm "$DAGGER_ENGINE_DOCKER_CONTAINER"
```

File: docs/current/guides/235290-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+1. Clear unused volumes and data:
+
+  ```shell
+  docker volume prune
+  docker system prune
+  ```
+
+:::tip
+If you are using a different container runtime, replace the commands above with the correct equivalents for your runtime.
+:::
```
Comment (gerhard) [inline docs/current/guides/235290-troubleshooting.md:40]: I would reword this to: `If you have custom provisioned the Dagger Engine, please adjust the above to your environments`.

There are multiple ways of running Dagger Engine, and not all of them involve a container runtime. Technically, Dagger Engine could be run on the host machine as a binary, outside of a container runtime. FWIW:

https://github.com/dagger/dagger/blob/2c04a71b0cdcbbe0e9712743a383f652a9f1211e/core/docs/d7yxc-operator_manual.md?plain=1#L140-L151

The default behaviour is `docker-image://<CONTAINER_IMAGE_URL>`.

File: docs/current/guides/235290-troubleshooting.md (hunk 2 above)
Comment (vikram-dagger) [reply docs/current/guides/235290-troubleshooting.md]: Added

File: docs/current/guides/235290-troubleshooting.md (hunk 3 above)
Comment (vikram-dagger) [reply docs/current/guides/235290-troubleshooting.md]: Done

**Review by gerhard — APPROVED** [review]
👍 🚀
